dotnet/src/SemanticKernel.Abstractions/SemanticKernel.Abstractions.csproj

```diff
@@ -23,6 +23,7 @@
     <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" />
     <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" />
     <PackageReference Include="System.Diagnostics.DiagnosticSource" />
+    <PackageReference Include="System.ClientModel" />
```

Comment:
If the purpose of adding `System.ClientModel` package is to use `BinaryData` type only (I can't find other types used, please correct me if I'm wrong), then you should probably use `System.Memory.Data` which contains `BinaryData` type:
https://www.nuget.org/packages/System.Memory.Data
![image](https://github.com/microsoft/semantic-kernel/assets/13853051/017694b2-850e-4776-a804-a10d3437ed38)

The reason why `BinaryData` is accessible with `System.ClientModel` is because `System.ClientModel` references `System.Memory.Data`.

But the purpose of `System.ClientModel` is different: `System.ClientModel contains building blocks for communicating with cloud services. It provides shared primitives, abstractions, and helpers for .NET service client libraries.`

Reply:
Azure library also references System.ClientModel...so we already are.  I thought it might be a more more efficient maintenance point to keep in sync with our existing dependency stack (avoid diverging)